Rubberduck.Core/UI/Command/Refactorings/RefactorImplementInterfaceCommand.cs

```diff
@@ -28,17 +28,19 @@ protected override bool EvaluateCanExecute(object parameter)
             var selection = Vbe.GetActiveSelection();        
 
             if (!selection.HasValue)
-                {
-                    return false;
-                }
+            {
+                return false;
+            }
 
-                var targetInterface = _state.AllUserDeclarations.FindInterface(selection.Value);
+            var targetInterface = _state.AllUserDeclarations.FindInterface(selection.Value);
 
-                var targetClass = _state.AllUserDeclarations.SingleOrDefault(d =>
-                    d.DeclarationType == DeclarationType.ClassModule &&
-                    d.QualifiedSelection.QualifiedName.Equals(selection.Value.QualifiedName));
+            var targetClass = _state.AllUserDeclarations.SingleOrDefault(d =>
+                d.DeclarationType == DeclarationType.ClassModule &&
+                d.QualifiedSelection.QualifiedName.Equals(selection.Value.QualifiedName));
```

Comment:
This could better be written as `_state.DeclarationFinder.UserDeclarations(DeclarationType.ClassModule).SingleOrDefault(declaration => declaration.QualifiedSelection.QualifiedName.Equals(selection.Value.QualifiedName))` or as `_state.DeclarationFinder.Members(selection.Value.QualifiedName).SingleOrDefault(declaration => declaration.DeclarationType == DeclarationType.ClassModule)`

Reply:
I went with the second one. I do assume in the general case it's quicker in reducing the samplesize checked with `SingleOrDefault`